client/pages/ConfirmPairing.jsx

```diff
@@ -67,16 +82,25 @@ export default class ConfirmPairing extends React.Component {
             activityFound: false
           });
         }
-      });
+      })
+      .catch(() => console.error('An unexpected error occurred'));
   }
 
   handleAccept() {
     const formData = this.state;
+    const savedUserDataJson = localStorage.getItem('userData');
+    let savedUserData = null;
+    let token = null;
+    if (savedUserDataJson !== null) {
+      savedUserData = JSON.parse(savedUserDataJson);
+      token = savedUserData.token;
+    }
```

Comment:
If you store your user and token in App's state and pass it via context, you can cut down on a lot of this duplication.